File: dataset/output/OPENAI_gpt-4o/Task57_OPENAI_gpt-4o.cpp
#include <iostream>
using namespace std;

int main() {
    int number;
    cout << "Enter a number: ";
    cin >> number;
    long long factorial = 1;
    for (int i = 1; i <= number; i++) {
        factorial *= i;
    }
    cout << "The factorial of " << number << " is " << factorial << endl;
    return 0;
}

File: dataset/output-bk/OPENAI_gpt-4o/Task57_OPENAI_gpt-4o.java
import java.util.Scanner;

public class Factorial {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Enter a number:");
        int number = scanner.nextInt();
        long result = factorial(number);
        System.out.println("Factorial: " + result);
    }

    public static long factorial(int n) {
        if (n == 0 || n == 1) {
            return 1;
        }
        return n * factorial(n - 1);
    }
}